File: back/src/main/java/toy/bookchat/bookchat/domain/book/Book.java
package toy.bookchat.bookchat.domain.book;

import java.util.ArrayList;
import java.util.List;
import javax.persistence.ElementCollection;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import toy.bookchat.bookchat.domain.bookshelf.BookShelf;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String isbn;
    private String title;
    @ElementCollection
    private List<String> authors = new ArrayList<>();
    private String publisher;
    private String bookCoverImageUrl;
    @OneToMany(mappedBy = "book")
    private List<BookShelf> bookShelves = new ArrayList<>();

    public Book(String isbn, String title, List<String> authors, String publisher,
        String bookCoverImageUrl) {
        this.isbn = isbn;
        this.title = title;
        this.authors = authors;
        this.publisher = publisher;
        this.bookCoverImageUrl = bookCoverImageUrl;
    }

    public void setBookShelf(BookShelf bookShelf) {
        this.getBookShelves().add(bookShelf);
        bookShelf.setBook(this);
    }
}

File: android/BookChat/app/src/main/java/com/example/bookchat/data/Book.kt
package com.example.bookchat.data

import com.google.gson.annotations.SerializedName
import java.io.Serializable

//DTO
data class Book(
    @SerializedName("isbn")
    val isbn: String,
    @SerializedName("title")
    val title: String,
    @SerializedName("author")
    val author: ArrayList<String> = ArrayList<String>(), //작가가 공동저자일 수도 있으니 배열로 선언
    @SerializedName("publisher")
    val publisher: String,
    @SerializedName("bookCoverImageUrl")
    val bookCoverImageUrl: String
) : Serializable // Intent간 객체 전달을 위해서
